### DISCUSSION TITLE
Does SpiceDB support Environment Variables for configuration?

### DISCUSSION THREAD

**Discussion by alessandromr** [authzed/spicedb#1495] [Q&A] [answered]
Hi,
I am following the project and I'm curious if you are planning to support env vars to simplify orchestrated deployments.
For a testing phase, it's already great to use docker and a custom entrypoint but when deploying to something like ECS/Kubernetes/Swarm it would be really great to set up the datastore connection using environment variables.

Given that you are already using viper it should be easy to add the support with something like AutomaticEnv(). 
Open also to start up a PR if it is something you agree with.

Semi related to: https://github.com/authzed/spicedb/issues/182

**Comment by jakedt** [comment, marked as answer]
You can already do this! We use a project called [cobrautil](https://github.com/jzelinskie/cobrautil) from one of our founders which binds to the environment in a `PreRunE`: 

https://github.com/jzelinskie/cobrautil/blob/2497cc69df22f338b15647b06111f12e63b8bd8b/cobrautil.go#L51

Then SpiceDB picks this up here:

https://github.com/authzed/spicedb/blob/2ea1f2ef69e6b35cd1bb5d3598e2f860730a9c23/cmd/spicedb/main.go#L8

You can set the value of any flag by specifying it in the form of:

`SPICEDB_<KEY_IN_ALL_CAPS_WITH_UNDERSCORES_INSTEAD_OF_DASHES>`

For example `--log-level` becomes `SPICEDB_LOG_LEVEL`.

Let us know if you have any additional questions. If you want to send a PR to update the documentation in a way which will be clear to you, we would be happy for the contribution!
